File: src/com/peyo/lbsettings/MenuItemPresenter.java
package com.peyo.lbsettings;

import android.content.Context;
import android.support.v17.leanback.widget.Presenter;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

public class MenuItemPresenter extends Presenter {
	
    private static class MenuItemViewHolder extends ViewHolder {
        public final ImageView mIconView;
        public final TextView mTitleView;

        MenuItemViewHolder(View v) {
            super(v);
            mIconView = (ImageView) v.findViewById(R.id.icon);
            mTitleView = (TextView) v.findViewById(R.id.title);
        }
    }

	@Override
	public void onBindViewHolder(ViewHolder viewHolder, Object item) {
	      if (item instanceof MenuItem && viewHolder instanceof MenuItemViewHolder) {
	            final MenuItem menuItem = (MenuItem) item;
	            MenuItemViewHolder menuItemViewHolder = (MenuItemViewHolder) viewHolder;
                menuItemViewHolder.mIconView.setImageResource(menuItem.getIconRes(
						menuItemViewHolder.mTitleView.getContext()));
                menuItemViewHolder.mTitleView.setText(menuItem.getTitleRes(
					  menuItemViewHolder.mTitleView.getContext()));
	        }
	}
	
	@Override
	public ViewHolder onCreateViewHolder(ViewGroup parent) {
        LayoutInflater inflater = (LayoutInflater) parent.getContext().getSystemService(
                Context.LAYOUT_INFLATER_SERVICE);
        View v = inflater.inflate(R.layout.browse_item, parent, false);
        return new MenuItemViewHolder(v);
	}

	@Override
	public void onUnbindViewHolder(ViewHolder viewHolder) {
        if (viewHolder instanceof MenuItemViewHolder) {
            MenuItemViewHolder menuItemViewHolder = (MenuItemViewHolder) viewHolder;
            menuItemViewHolder.mIconView.setImageBitmap(null);
        }
	}
	
}

File: src/com/peyo/lbsettings/MenuItemPresenter.kt
package com.peyo.lbsettings

import android.content.Context
import androidx.leanback.widget.Presenter
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView

class MenuItemPresenter : Presenter() {

    private class MenuItemViewHolder internal constructor(v: View) : Presenter.ViewHolder(v) {
        val mIconView: ImageView
        val mTitleView: TextView

        init {
            mIconView = v.findViewById(R.id.icon) as ImageView
            mTitleView = v.findViewById(R.id.title) as TextView
        }
    }

    override fun onBindViewHolder(viewHolder: Presenter.ViewHolder, item: Any) {
        if (item is MenuItem && viewHolder is MenuItemViewHolder) {
            viewHolder.mIconView.setImageResource(item.getIconRes(
                    viewHolder.mTitleView.context))
            viewHolder.mTitleView.setText(item.getTitleRes(
                    viewHolder.mTitleView.context))
        }
    }

    override fun onCreateViewHolder(parent: ViewGroup): Presenter.ViewHolder {
        val inflater = parent.context.getSystemService(
                Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
        val v = inflater.inflate(R.layout.browse_item, parent, false)
        return MenuItemViewHolder(v)
    }

    override fun onUnbindViewHolder(viewHolder: Presenter.ViewHolder) {
        if (viewHolder is MenuItemViewHolder) {
            viewHolder.mIconView.setImageBitmap(null)
        }
    }

}
